### ISSUE TITLE
SimplyImprovedTerrain crashes RS's End Mineshaft

### ISSUE THREAD

**Issue by PseudoDistant** [TelepathicGrunt/RepurposedStructures-Fabric#135]
I have a lot of mods on the server, but I have narrowed it down to RepurposedStructures.
(I've confirmed by removing RepurposedStructures and testing without it.)
Here's the latest crash log:
[crash-2021-07-30_14.15.57-server.txt](https://github.com/TelepathicGrunt/RepurposedStructures-Fabric/files/6909363/crash-2021-07-30_14.15.57-server.txt)
(I'm not sure if it's a conflict or not, I'll test it more later.)
Please tell me if it's something I did wrong, as I am not unwilling to accept that I did something stupid, I really need this to work.

**Comment by TelepathicGrunt** [comment]
I'm not seeing anything that points to any issue. Can you export your modpack so i can try it myself? Or can you download the mod called Blame, enter the end, crash, and the retrieve the latest.log file from the log folder?  The latest.log may have more info than the crashlog.

**Comment by PseudoDistant** [comment]
[latest.log](https://github.com/TelepathicGrunt/RepurposedStructures-Fabric/files/6909908/latest.log)

**Comment by PseudoDistant** [comment]
Server mod files:
```
AxesAreWeapons-1.3-1.16.5.jar
BetterMineshafts-Fabric-1.17-1.0.0.jar
DimThread-1.2.5.jar
Fireplace-Lib-5.2.2+1.17.1.jar
FiresSurvivalTweaks-2.2.0+1.17.1.jar
Geyser-Fabric-1.4.0-SNAPSHOT.jar
Harvest-fabric-1.16.1-1.0.1.jar
Lakeside-1.1.0+1.17.jar
LessAnnoyingPiglins-1.1.0.jar
LuckPerms-Fabric-5.3.53.jar
OverpoweredMending-1.17-2.2.3.jar
PerfectAccuracy-1.1.jar
SimplyImprovedTerrain-0.4.0.2-Fabric-1.17.jar
YungsApi-1.17-Fabric-13.jar
appleskin-fabric-mc1.17-2.0.0.jar
beaconsforall-fabric-2.0.0-1.17.jar
bedrockwaters-1.5.0+1.17.jar
better-signs-and-frames-0.8.2.jar
betterbeacon-1.0.2.jar
blame-1.17-4.0.0-fabric.jar
camp-fires-cook-mobs-1.1.10+MC1.17-1.17.1.jar
carpet-extra-1.17.1-1.4.43.jar
castle_dungeons-2.6-1.17-fabric.jar
cavenabler-1.0.0.jar
chicken-nerf-1.0.7+MC1.17-1.17.1.jar
classic-beaches-0.3.1.jar
crowmap-1.17-1.0.1.jar
economical-villager-trading-1.2+MC1.17.jar
eggofcapitalism-mc1.17.x-2.1.jar
expanded-axe-enchanting-1.0.2.jar
expanded-trident-enchanting-1.0.2.jar
fabric-api-0.36.1+1.17.jar
fabric-carpet-1.17.1-1.4.43+v210721.jar
fabric-language-kotlin-1.6.2+kotlin.1.5.20.jar
fishybears-0.1.jar
floodgate-fabric.jar
flytre_lib-1.17-0.2.21.jar
giant_ai-1.17.1-1.1.0.jar
harmfulstonecutters-1.0.2.jar
immersive-cursedness-1.4.4.jar
infinity-water-bucket-1.4-mc1.17-fabric.jar
krypton-0.1.4.jar
lapisreserve-1.0.8.jar
lazydfu-0.1.2.jar
leads-0.6.5+1.17-patched.jar
leads-0.6.5+1.17.jar.disabled
lengthyladders-1.0.4.jar
lithium-fabric-mc1.17.1-0.7.3.jar
lucid-lite-1.0.0.jar
mobs-attempt-parkour-0.2.3-1.17-pre1.jar
mostructures-1.2.1-1.17.jar
no-teleport-cooldown-1.0.2.jar
powershot-1.2.0-1.17-fabric.jar
projecthotbreak-1.2.005.jar
recipecache-0.2.0-1.17.1.jar
repurposed_structures_fabric-2.1.7+1.17.1.jar
slipperywalls-1.0.0+1.17.jar
smoothboot-fabric-1.16.5-1.6.0.jar
snowballs-hit-players-1.0.0.jar
starlight-1.0.0-RC3+fabric.63660a0.jar
suitablystackablestew-1.1.2.jar
tabtps-fabric-mc1.17-1.3.8.jar
the_loved_ones-1.2.0+1.17.jar
vanish-1.17.1-1.2.6.jar
villagers-follow-emeralds-1.1.0.jar
worldedit-fabric-mc1.17-7.2.6-beta-01.jar
```
(Some of the mods are 1.16 on CurseForge, but have working 1.17 branches on GitHub.)
(Leads is patched to allow itself to run on Java 17, since TabTPS needs Java 17 to work correctly with my system.)
https://github.com/jpenilla/TabTPS/issues/22

**Comment by PseudoDistant** [comment]
I think it might be a mod conflict, but I don't know what mod would be causing the problem, or why it's only causing problems in The End for that matter.

**Comment by TelepathicGrunt** [comment]
This is the crash mapped
https://paste.ee/p/LzgI8

My method that I call is this
https://github.com/TelepathicGrunt/RepurposedStructures-Fabric/blob/766498c5a4c55aed70c144347c6d6bbffb4a93a9/src/main/java/com/telepathicgrunt/repurposedstructures/world/structures/MineshaftEndStructure.java#L44

Whatever mod is the issue, it is making the internals of `chunkGenerator.getFirstOccupiedHeight` to crash in a way that I have no seen before and not sure how it was accomplished. 

Does it still crash if you remove lithium or SimplyImprovedTerrain?

**Comment by TelepathicGrunt** [comment]
Got it. It is SimplyImprovedTerrain crashing my mod. I'll contact the dev

**Comment by PseudoDistant** [comment]
> Got it. It is SimplyImprovedTerrain crashing my mod. I'll contact the dev

Good to know. :D
I'll be watching for SIT updates, and in the meanwhile I guess I'll just disable The End, thank you. :)

**Comment by PseudoDistant** [comment]
Also sorry I wasn't available for a while, I was doing laundry stuff.

**Comment by TelepathicGrunt** [comment]
looks like it was fixed in v0.4.0.3 simply improved terrain. I almost forgot about this issue lol. And I didnt realized an update came out for it. i'll close this now
